### DISCUSSION TITLE
Hacks we would like to fund // grants

### DISCUSSION THREAD

**Discussion by yangwao** [kodadot/grants#2] [General]
Once we have API for reading data https://github.com/kodadot/packages/tree/main/uniquery and submitting on-chain actions, we would like to see more hacks built on top of our v0 API

You are more than welcome to submit ideas you want to get funded. 

Feel free to reach out yangwao on KodaDot Discord for more clarification 

# Request for proposals
- https://github.com/kodadot/grants/issues/5
- bot to help manage and make offers on Basilisk - https://github.com/kodadot/packages/issues/55  
- floor sweeping application (PR to KodaDot / API / bot)
- implementing replicate.com API to mint from KodaDot directly 
  - https://github.com/kodadot/nft-gallery/issues/3455	
- Generative art frameworks https://github.com/kodadot/nft-gallery/discussions/3335
- https://github.com/kodadot/grants/issues/6
- https://github.com/kodadot/grants/issues/7
- https://github.com/kodadot/grants/issues/8 
- https://github.com/kodadot/grants/issues/16 
- https://github.com/kodadot/grants/discussions/3 
- MysterMints -- for lazy minting and adding traits, users submitting tx on users' behalf?

## Edit
Most probably I will move it to separate repository to drive this, discussion is nice place to bootstrap things but in long term efforts might get messy in terms of process and tracking stuff

**Comment by roiLeo** [comment]
Some ideas
- [ ] nft candy machine project
- [ ] rmrk 2 integration -> https://github.com/kodadot/nft-gallery/issues/3725

Non-related
- [ ] XR app
- [ ] swap/stacking project


Or I may not be aware of futur roadmap.

**Comment by damskyftw** [comment]
After chit chat with @yangwao, we’d like to fund certain form of POAP minter on KodaDot.

We’d take our simple mint with advanced airdrop function (which works) and simply create new UI which would use simple function minter.

The POAP minter should be/have:

- Easy to use
- Claim function
- Low to zero fees
- ?
- Airdrop function, if addresses collected
- If not, just send link through which you can claim your stuff


Related 

- https://github.com/kodadot/nft-gallery/issues/3668

**Comment by dtbx** [comment]
### NFT powered DeFi

_Question:_ 
How can we introduce a new vertical of revenue to KodaDot?

_Answer:_
Build NFT collateralization protocol with Acala aUSD and incentivise it in the KodaDot and greater DotSama ecosystem.


_Context:_ 
It's been long known that liquidity is the lifeblood of markets, including crypto markets. However, NFTs, by virtue of being non fungible, are low(er) velocity assets (low velocity: financial jargon referring to assets that typically change hands slow) this also means appraisal/price discovery is inefficient. 

This situation is not ideal for pretty much all stakeholders of the NFT ecosystem: protocols, creators, collectors the marketplace/platform. 

_Solution:_ 
Introduce DeFi incentives to increase NFTs utility, improve assets price discovery, increase platform and protocols associated revenue.

The simplest product that could do that is NFT lending similar to [NFTfi](https://www.nftfi.com). 

**How does it work in a nutshell:?**

It's a lending and borrowing product: 

Lender: 
Provide fungible assets against collateral, e.g. volatile assets such as DOT, KSM or fiat pegged assets like Acala aUSD or Karura kUSD.

Borrower: 
NFT owner provides NFTs as collateral and takes fungible assets as an interest bearing loan. 

Going with similar example to NFT.fi loans

1. Borrower lists an _NFT_ 
2. Lenders make offers (loan amount, loan period, interest rate)
3. Borrowers accept desired offer

Two possible scenarios

4. Borrower pays back the loan + interest rate before end of loan period
5. Borrower default: NFT remains with the lender, principle remains with the borrower. 

Potential partners:

- DeFi asset issuer such as Acala through their grants in order to incentivise both lenders and borrowers: [https://acala.network/ecosystem-program](https://acala.network/ecosystem-program). I haven't checked yet if it's still running though.
- Propose to fund this project and incentives associated to it via Polkadot council or Kusama treasury. I think it could qualify as it's experimentation in public good rather than for profit at this stage.


Consideration #1: I'd seek to understand more if NFT.fi worked and where did they have challenges, potential innovations that they introduced after testing their hypothesis. 

Considerations #2: It's bear market and most of the assets on KodaDot atm aren't super high value, it means identifying certain amount of sufficiently valuable assets and interest in them will be needed order to kickstart this market.

Consideration #3: it's a project that will require builders + business acumen. I'd consider pricing it with $50k-100k spread across 3-4 milestones. Could very well turn into an investible project.

Reference reading: [Looking for Liquidity? Here’s Everything to Know About NFT Lending
BY RANDY GINSBURG •
AUGUST 18, 2022](https://nftnow.com/guides/looking-for-liquidity-heres-everything-to-know-about-nft-lending/)

**Comment by yangwao** [comment]
Pick up some game ideas. Then reach out to me 

## Web3 Game Suggestions
- https://edward-thomson.medium.com/subrogue-request-for-help-cb770dd83eac
- https://edward-thomson.medium.com/subrogue-the-build-so-far-ca39edb4edc3
- https://edward-thomson.medium.com/on-infinite-multi-user-dungeons-4ef13d275a35
- https://edward-thomson.medium.com/subrogue-a-simple-game-with-tough-problems-87748fe2839c
- https://edward-thomson.medium.com/subrogue-8c0a537f02d4
- https://edward-thomson.medium.com/the-worlds-of-web-3-games-7e4797126a94
- https://edward-thomson.medium.com/blockchain-gaming-putting-the-state-on-chain-cc3915090547

**Comment by yangwao** [comment]
- Ticketing service - few more people are interested

**Comment by damskyftw** [comment]
## Idea dump
Custom gaming world based on [Epic games concept ](https://youtu.be/ePcZLfty3ag)

A possible use case for the gaming sector in Web3 combining NFTs and shared revenue between players could involve creating a platform that allows players to design and develop their own games, collaborate with others, and earn revenue from their creations. Players could create unique in-game assets, such as characters, weapons, and maps, and tokenize them as NFTs. These NFTs could be used across multiple games within the platform, adding value and versatility to the assets. The platform could also implement a shared revenue model, where a percentage of the revenue generated from microtransactions, such as cosmetic purchases, is distributed among the creators of the games based on their popularity and engagement. This would incentivize creators to develop high-quality, engaging content and foster a competitive environment, ultimately leading to a diverse and thriving gaming ecosystem.

## What would need to be done?


- Built on Polkadot / custom parachain?

- Develop a token standard: Create a token standard for the NFTs that will be used in the platform. This standard should be compatible within polkadot and include necessary attributes like unique identifiers, ownership, and metadata.

- Create a smart contract framework: Design and develop smart contracts to manage the creation and trading of NFTs, the shared revenue model, and other game-related functions. This may include contracts for minting NFTs, distributing revenue, and tracking player data.

- Design a user interface: Develop a user-friendly interface for the platform, allowing players and creators to easily interact with the smart contracts, create and manage their NFTs, and access the various games on the platform.

- Implement cross-game compatibility: Establish a system that enables NFTs to be used seamlessly across multiple games within the platform. This may involve creating a shared database of NFT attributes or developing a standard for game developers to follow.

- Build a marketplace: Create a marketplace where players can buy, sell, and trade their NFTs. This marketplace should be integrated with the platform's smart contracts to ensure secure and transparent transactions.

- Develop a revenue distribution model: Design a fair and transparent model for distributing revenue generated from microtransactions among game creators. The model should consider factors like popularity and engagement, and be implemented using smart contracts.

- Provide developer tools and resources: Offer tools, SDKs, and documentation to help game developers easily create and integrate their games with the platform. This will encourage a diverse range of games and foster a thriving ecosystem.

- Ensure security and scalability: Continuously monitor and update the platform's security measures to protect against potential threats. Additionally, plan for the platform's scalability to accommodate increasing numbers of users, NFTs, and games over time.

### Next steps

- Research on similar projects and solutions

**Comment by 0xslipk** [comment]
Hi guys,

I am the lead developer of [Blockcoders](https://blockcoders.io/). We are a is a highly skilled software development company specializing in web3 projects and blockchain platforms. With a mission to unlock the full potential of web3 technology, Blockcoders aims to empower builders, developers, and makers by providing them with easy-to-use tools and resources.

We are currently looking for projects to collaborate with and Kodadot picked our attention. I can see that you guys have a few things for RFP. Do you have a priority that you need a team to work on or what idea are deprecated?

Best,

**Reply by yangwao** [reply]
Hey, currently our grants works for you if you delivered at least some PoW towards our ecosystem. 

Can you point me on your merged PRs in one or our project repositories to evaluate if you're a good fit?

We are  currently not looking for dev studios, but group of individuals.

**Reply by 0xslipk** [reply]
Hey, nice to meet you.

As you can see [here](https://github.com/w3f/Grants-Program/pulls?q=is%3Apr+is%3Aclosed+author%3A0xslipk). We have experience in the substrate ecosystem.

We are not a dev studios. We are just a group of dev working on blockchain. We usualy work applying for grants :D
